pyiceberg/table/__init__.py

```diff
@@ -944,8 +944,8 @@ def snapshot(self) -> Optional[Snapshot]:
     def projection(self) -> Schema:
         snapshot_schema = self.table.schema()
         if snapshot := self.snapshot():
-            if snapshot_schema_id := snapshot.schema_id:
-                snapshot_schema = self.table.schemas()[snapshot_schema_id]
+            snapshot_schema_id = snapshot.schema_id
+            snapshot_schema = self.table.schemas()[snapshot_schema_id]
```

Comment:
Thanks for opening this PR @tonyps1223!

The schema-id on the snapshot is [optional](https://iceberg.apache.org/spec/#snapshots), I think we still need to check if it is non-null, but make sure that we still pass on `0`:
```suggestion
            if snapshot.schema_id is not None:
                snapshot_schema = self.table.schemas()[snapshot.schema_id]
```